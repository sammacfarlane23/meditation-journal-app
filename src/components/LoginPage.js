import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className='box-layout'>
    <div className='box-layout__box'>
      <h1 className='box-layout__title'>Meditation Journal</h1>
      <p>A simple app to journal your experience of meditation</p>
      <button className='button button--login' onClick={startLogin}>
        Login with Google
      </button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
