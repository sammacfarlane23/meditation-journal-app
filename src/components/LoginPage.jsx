import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startFacebookLogin }) => (
  <div className='box-layout'>
    <div className='box-layout__box'>
      <h1 className='box-layout__title'>Meditation Journal</h1>
      <p>A simple app to journal your experience of meditation</p>
      <div className='login-button-container'>
        <button
          className='button button--login-google'
          onClick={startGoogleLogin}
        >
          Login with Google
        </button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
