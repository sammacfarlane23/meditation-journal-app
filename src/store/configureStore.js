import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import authReducer from "../reducers/auth";
import entriesReducer from "../reducers/entries";
import filtersReducer from "../reducers/filters";

const configuredStore = () => {
  const store = createStore(
    combineReducers({
      entries: entriesReducer,
      auth: authReducer,
      filters: filtersReducer,
    }),
    applyMiddleware(thunk)
  );

  return store;
};

export default configuredStore;
