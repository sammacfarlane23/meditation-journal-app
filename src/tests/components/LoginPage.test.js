import React from 'react';
import LoginPage from '../../components/LoginPage';
import { renderWithState } from '../store/mockStore';

test('should render login page correctly', () => {
  const component = renderWithState(<LoginPage />);
  expect(component).toMatchSnapshot();
});
