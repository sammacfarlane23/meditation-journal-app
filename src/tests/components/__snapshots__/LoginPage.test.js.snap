// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Meditation Journal
    </h1>
    <p>
      A simple app to journal your experience of meditation
    </p>
    <div
      className="login-button-container"
    >
      <button
        className="button button--login-google"
      >
        Login with Google
      </button>
    </div>
  </div>
</div>
`;
